internal/resource-request-operator/utils.go

```diff
@@ -51,3 +52,29 @@ func (r *ResourceRequestReconciler) generateResourceOffer(request *discoveryv1al
 	klog.Infof("%s -> %s Offer: %s", r.ClusterID, op, offer.ObjectMeta.Name)
 	return nil
 }
+
+func (r *ResourceRequestReconciler) createForeignCluster(resourceRequest *discoveryv1alpha1.ResourceRequest) error {
+	foreignCluster := &discoveryv1alpha1.ForeignCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: resourceRequest.Spec.ClusterIdentity.ClusterID,
+			Labels: map[string]string{
+				discovery.DiscoveryTypeLabel: string(discovery.IncomingPeeringDiscovery),
+				discovery.ClusterIDLabel:     resourceRequest.Spec.ClusterIdentity.ClusterID,
+			},
+		},
+		Spec: discoveryv1alpha1.ForeignClusterSpec{
+			ClusterIdentity: resourceRequest.Spec.ClusterIdentity,
+			Namespace:       resourceRequest.Namespace,
+			Join:            false,
+			DiscoveryType:   discovery.IncomingPeeringDiscovery,
+			AuthURL:         resourceRequest.Spec.AuthURL,
+		},
+	}
```

Comment:
@aleoli Is it sufficient to create this foreigncluster or its definition should also be enforced in case of modifications?

Reply:
It should be enforced, but only if the discovery type is still IncomingPeering, since it can be discovered in other ways (i.e. mDNS, DNS) that can overwrite this method and change its content.
And then we have to take care and not enforce the join flag, to preserve user chooses.

Reply:
Got it. As agreed, for the moment it is probably better not to enforce it to avoid problems, since all fields should be anyway immutable (at least conceptually)